### PR TITLE
Port fix of tapyrus crash from master branch to v0.5 branch

### PR DESCRIPTION
**Pull request by Naviabheeman**
Port fix from issue #254 from master branch to v0.5 branch along with build setting changes.

- CI runs on ubuntu20
- macOS sdk is 10.14. and Xcode 10.1

### CODE REVIEW

File: doc/README_osx.md [hunk 1]
```diff
@@ -41,19 +41,19 @@ To create a tarball suitable for Gitian input, there are two options:
...
   $ hdiutil attach Xcode_7.3.1.dmg
-  $ tar -C /Volumes/Xcode/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/ -czf MacOSX10.11.sdk.tar.gz MacOSX10.11.sdk
+  $ tar -C /Volumes/Xcode/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/ -czf MacOSX10.14.sdk.tar.gz MacOSX10.14.sdk
 ```
 
 Alternatively, you can use 7zip and SleuthKit to extract the files one by one.
 The script contrib/macdeploy/extract-osx-sdk.sh automates this. First ensure
 the dmg file is in the current directory, and then run the script. You may wish
-to delete the intermediate 5.hfs file and MacOSX10.11.sdk (the directory) when
+to delete the intermediate 5.hfs file and MacOSX10.14.sdk (the directory) when
```
Comment (azuchi) [inline doc/README_osx.md:50]: This change also need for master branch.

File: src/validation.cpp [hunk 2]
```diff
@@ -4286,8 +4286,11 @@ bool LoadExternalBlockFile(FILE* fileIn, CDiskBlockPos *dbp, std::vector<XFieldA
                     while (range.first != range.second) {
                         std::multimap<uint256, CDiskBlockPos>::iterator it = range.first;
                         std::shared_ptr<CBlock> pblockrecursive = std::make_shared<CBlock>();
+                        BlockMap::iterator miSelf = mapBlockIndex.find(hash);
+                        CBlockIndex *pindex = nullptr;
+                        pindex = miSelf->second;
 
-                        if (ReadBlockFromDisk(*pblockrecursive, it->second, pblockrecursive->GetHeight()))
+                        if (ReadBlockFromDisk(*pblockrecursive, it->second, pindex->nHeight))
```
[the remark this answers is not in the document]
Comment (Naviabheeman) [reply src/validation.cpp]: I rebased it.

File: Makefile.am [hunk 3]
```diff
@@ -81,6 +81,7 @@ $(BITCOIN_WIN_INSTALLER): all-recursive
 	  echo error: could not build $@
 	@echo built $@
```
Comment (azuchi) [inline Makefile.am:83]: Why is this change necessary for the 0.5 branch? In addition, the following PR indicates that changes to this file alone are not sufficient.

https://github.com/bitcoin/bitcoin/pull/28432

**Review by azuchi — CHANGES_REQUESTED** [review]

File: Makefile.am (hunk 3 above)
Comment (Naviabheeman) [reply Makefile.am:83]: I realise what happened. The CI had completed because it skipped the packaging step. I was so relieved about its success that it did not even occur to me to check the logs. 
There are gaps in macOS packaging. I'm fixing them.

File: depends/packages/native_mac_alias.mk [hunk 4]
```diff
@@ -1,15 +0,0 @@
...
-$(package)_install_libdir=$(build_prefix)/lib/python/dist-packages
-
-define $(package)_build_cmds
-    python setup.py build
-endef
-
-define $(package)_stage_cmds
-    mkdir -p $($(package)_install_libdir) && \
-    python setup.py install --root=$($(package)_staging_dir) --prefix=$(build_prefix) --install-lib=$($(package)_install_libdir)
-endef
```
Comment (azuchi) [inline depends/packages/native_mac_alias.mk:15]: Why delete this? Is it still in the master branch?

File: depends/packages/native_ds_store.mk [hunk 5]
```diff
@@ -1,16 +0,0 @@
...
-$(package)_dependencies=native_biplist
-
-define $(package)_build_cmds
-    python setup.py build
-endef
-
-define $(package)_stage_cmds
-    mkdir -p $($(package)_install_libdir) && \
-    python setup.py install --root=$($(package)_staging_dir) --prefix=$(build_prefix) --install-lib=$($(package)_install_libdir)
-endef
```
Comment (azuchi) [inline depends/packages/native_ds_store.mk:16]: Why delete this? Is it still in the master branch?

File: contrib/macdeploy/gen-sdk [hunk 6]
```diff
@@ -81,7 +81,7 @@ def run():
 
     print("Creating output .tar.gz file...")
     with out_sdktgz_path.open("wb") as fp:
-        with gzip.GzipFile(fileobj=fp, compresslevel=9, mtime=0) as gzf:
+        with gzip.GzipFile(fileobj=fp, mode='wb', compresslevel=9, mtime=0) as gzf:
```
Comment (azuchi) [inline contrib/macdeploy/gen-sdk:84]: Why is this change necessary?

Comment (Naviabheeman) [reply contrib/macdeploy/gen-sdk:84]: This is a port from bitcoin commit 655d52a. I was made while attempting to upgrade macdeploy and its related script. Its not necessary in tapyrus.

**Review by azuchi — APPROVED** [review]
